src/game/behaviors/water_ring.inc.c

```diff
@@ -199,7 +210,13 @@ void manta_water_ring_act_not_collected(void) {
     if (o->oTimer > 150) {
         o->oOpacity -= 2;
         if (o->oOpacity < 3) {
-            o->activeFlags = ACTIVE_FLAG_DEACTIVATED;
+            if ((o->oWaterRingIndex == (ringManager->oWaterRingMgrLastRingCollected + 1))
+             && (ringSpawner->oWaterRingSpawnerRingsCollected < 5)
+             && (ringManager->oWaterRingSpawnerRingsCollected < 5)) {
+                ringSpawner->oWaterRingSpawnerRingsCollected = 0;
+                ringManager->oWaterRingSpawnerRingsCollected = 0;
+            }
+            obj_mark_for_deletion(o);
```

Comment:
sorry if you've already explained this but whats the reason for the change here? i dont think it has anything to do with the water ring fix and its not ifdef wrapped anyways

Reply:
This resets the ring counter when the next active ring is missed and disappears. This allows Mario to immediately try again instead of having to swim through another ring to reset the counter. You're right this is an unrelated change to the collision fix and should probably be in a separate PR and ifdef.